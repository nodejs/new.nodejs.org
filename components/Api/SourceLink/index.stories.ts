import SourceLink from './index';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';

type Story = StoryObj<typeof SourceLink>;
type Meta = MetaObj<typeof SourceLink>;

export const Default: Story = {
  args: {
    version: '1.0.0',
    link: 'http://nodejs.org/version/1.0.0',
  },
};

export default { component: SourceLink } as Meta;
