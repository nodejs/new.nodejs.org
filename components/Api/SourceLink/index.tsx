import { FormattedMessage } from 'react-intl';

import styles from './index.module.scss';
import type { FC } from 'react';

type SourceLinkProps = {
  link: string;
  version: string;
};

const SourceLink: FC<SourceLinkProps> = ({ version, link }) => (
  <p className={styles.sourceLinkComponent}>
    <FormattedMessage id="components.api.sourceLink" />{' '}
    <a href={`https://github.com/nodejs/node/blob/${version}/${link}`}>
      {link}
    </a>
  </p>
);

export default SourceLink;
