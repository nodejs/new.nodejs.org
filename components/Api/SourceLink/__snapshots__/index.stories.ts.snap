// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api/SourceLink Default smoke-test 1`] = `
<p class="SourceLink_sourceLinkComponent__1vzdY">
  Source Code:
  <a href="https://github.com/nodejs/node/blob/1.0.0/http://nodejs.org/version/1.0.0">
    http://nodejs.org/version/1.0.0
  </a>
</p>
`;
